sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,205 @@
+
+    Parameters
+    ----------
+    xx0 : ndarray of shape (grid_resolution, grid_resolution)
+        First output of :func:`meshgrid <numpy.meshgrid>`.
+
+    xx1 : ndarray of shape (grid_resolution, grid_resolution)
+        Second output of :func:`meshgrid <numpy.meshgrid>`.
+
+    response : ndarray of shape (grid_resolution, grid_resolution)
+        Values of the response function.
+
+    Attributes
+    ----------
+    surface_ : matplotlib `QuadContourSet` or `QuadMesh`
+        If `plot_method` is 'contour' or 'contourf', `surface_` is a
+        :class:`QuadContourSet <matplotlib.contour.QuadContourSet>`. If
+        `plot_method is `pcolormesh`, `surface_` is a
+        :class:`QuadMesh <matplotlib.collections.QuadMesh>`.
+
+    ax_ : matplotlib Axes
+        Axes with confusion matrix.
+
+    figure_ : matplotlib Figure
+        Figure containing the confusion matrix.
+    """
+    def __init__(self, xx0, xx1, response):
+        self.xx0 = xx0
+        self.xx1 = xx1
+        self.response = response
+
+    def plot(self, plot_method='contourf', ax=None, **kwargs):
+        """Plot visualization.
+
+        Parameters
+        ----------
+        plot_method : {'contourf', 'contour', 'pcolormesh'}, default='contourf'
+            Plotting method to call when plotting the response. Please refer
+            to the following matplotlib documentation for details:
+            :func:`contourf <matplotlib.pyplot.contourf>`,
+            :func:`contour <matplotlib.pyplot.contour>`,
+            :func:`pcolomesh <matplotlib.pyplot.pcolomesh>`.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be pased to the `plot_method`.
+
+        Returns
+        -------
+        display: :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+        """
+        check_matplotlib_support('DecisionBoundaryDisplay.plot')
+        import matplotlib.pyplot as plt  # noqa
+
+        if plot_method not in ('contourf', 'contour', 'pcolormesh'):
+            raise ValueError("plot_method must be 'contourf', 'contour', or "
+                             "'pcolormesh'")
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        plot_func = getattr(ax, plot_method)
+        self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_decision_boundary(estimator, X, grid_resolution=100, eps=1.0,
+                           plot_method='contourf', response_method='auto',
+                           ax=None, **kwargs):
+    """Plot Decision Boundary.
+
+    Please see examples below for usage.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Trained estimator.
+
+    X : array-like of shape (n_samples, 2)
```

Comment:
Since we are passing `X`, maybe we could have an option to make the `scatter` plot at the same time and to set the x_min/x_max/y_min/y_max limits. Basically, I think that most of the time, we plotting both the boundary and the scatter plot.

Reply:
I included this at the start, but I felt like it inflated the API. As in, we may need a `scater_kwargs` in the `plot_decisions_boundary`.

In its current form, plotting the scatter plot isn't too bad:

```py
fig, ax = plt.subplots()
plot_decision_boundary(est, X, y, ax=ax)
ax.scatter(X[:, 0], X[:, 1], ...)
```

It's common to want to adjust the alpha when there are too many points (or subsample).

Reply:
I think that it would be fine to have `scatter_kwargs` in fact, I would like to have to write the minimum lines of code, especially `xlim`, `ylim`, `xlabel`, `ylabel`, `alpha`, etc., for the largest number of our examples. Then, when one wants to fine-tune it will come back as efficient to write some matplotlib.